sklearn/metrics/cluster/unsupervised.py

```diff
@@ -47,11 +52,25 @@ def silhouette_score(X, labels, metric='euclidean', sample_size=None,
         <sklearn.metrics.pairwise.pairwise_distances>`. If X is the distance
         array itself, use ``metric="precomputed"``.
 
+    method: {'global', 'blockwise'}
+        The method used to compute distance matrix between samples. Default is
+        ``global`` which means that the full distance matrix is computed
+        yielding in fast computation but high memory consumption. The
+        ``blockwise`` option computes clusterwise distance matrices, dividing
+        memory consumption by approximately the squared number of clusters.
+        The ``blockwise`` method allows parallelization through ``n_jobs``
+        parameter.
```

Comment:
I suggest that we rename this argument to 'blockwise', and have as options False, True, and 'auto', where 'auto' is the default, and you do some clever check on the size of the input to decided whether blocking is good or not.